### ISSUE TITLE
[Suggestions] .addAttackEffect

### ISSUE THREAD

**Issue by RobertHDPotatoes** [Witixin1512/ArmorSetEffects#8]
1: .addAttackEffect
So, .addAttackerEffect and .addEffect already exists. However those leave a gap where a player cannot give a potion effect on attack, so I suggest adding .addAttackEffect, which will make it so if you have the specified equipment, then the player's attacks give a potion effect. For example, 
https://pastebin.com/dt2yPvUU
That script would give any entity attacked by a player wielding a wooden sword the effect poison, for 60 ticks. This could also work well with specific armor as well, but my example is very basic to get my point across.

2: Controllable ArmorSetEffect refresh time
I know you already mentioned how most of the logic would have to be reimplemented to allow for this to work (I'm gonna be honest, I'm really ignorant of modding in general so I don't really know what this means besides a lot of work lol), but I still think it should be considered at least. Currently, ArmorSetEffects only reloads every 5 seconds. This is fine for most situations, especially armor. However, if you use .inSlot(<equipmentslottype:mainhand>, it refreshes too slow to work consistently during gameplay. If you wanted to have some set effects when holding a sword, there are a lot of times you may not have your sword out for a full 5 seconds, or may switch to another item right at the specific time the set effect was supposed to reload, making you miss out on 5+ seconds of effects. I have no idea how hard this would be to implement, I'm really bad with mods and stuff, but I do think this would allow a lot more flexibility. My suggestion is to make it default to refresh every 5 seconds unless otherwise stated, but if stated it would respect the time you set. The reason I suggest that is because 5 seconds is fine for armor, but you may want a faster rate for main hand or off hand items.

Hopefully my suggestions make sense and are feasible. Thanks for upkeeping and porting this mod!

**Comment by Witixin1512** [comment]
I just posted a new file but the changelog is basically this:
Added `addAttackEffect` and `addAttackedEffect`, first one triggers when attacking, other one triggers when attacked.
They both take an MCPotionEffectInstance parameter, just like add `addEffect`
I changed the intervals of the the code to be null, so it runs every tick now
Let me know if there's anything else!

Also rewriting it wouldn't be _that_ hard, but it would require rewriting a tiny part of the mod that I would rather avoid touching. If possible, the interval at which armor sets are checked is global and untouchable
